crates/oxc_linter/src/options/filter.rs

```diff
@@ -0,0 +1,272 @@
+            let (plugin, rule) = match filter {
+                Cow::Borrowed(filter) => {
+                    let mut parts = filter.splitn(2, '/');
+
+                    let plugin = parts
+                        .next()
+                        .ok_or(InvalidFilterKind::PluginMissing(Cow::Borrowed(filter)))?;
+                    if plugin.is_empty() {
+                        return Err(InvalidFilterKind::PluginMissing(Cow::Borrowed(filter)));
+                    }
+
+                    let rule = parts
+                        .next()
+                        .ok_or(InvalidFilterKind::RuleMissing(Cow::Borrowed(filter)))?;
+                    if rule.is_empty() {
+                        return Err(InvalidFilterKind::RuleMissing(Cow::Borrowed(filter)));
+                    }
+
+                    (LintPlugins::from(plugin), Cow::Borrowed(rule))
+                }
+                Cow::Owned(filter) => {
+                    let mut parts = filter.splitn(2, '/');
+
+                    let plugin = parts
+                        .next()
+                        .ok_or_else(|| InvalidFilterKind::PluginMissing(filter.clone().into()))?;
+                    if plugin.is_empty() {
+                        return Err(InvalidFilterKind::PluginMissing(filter.into()));
+                    }
+
+                    let rule = parts
+                        .next()
+                        .ok_or_else(|| InvalidFilterKind::RuleMissing(filter.clone().into()))?;
+                    if rule.is_empty() {
+                        return Err(InvalidFilterKind::RuleMissing(filter.into()));
+                    }
+
+                    (LintPlugins::from(plugin), Cow::Owned(rule.to_string()))
+                }
+            };
+            Ok(LintFilterKind::Rule(plugin, rule))
+        } else {
+            match RuleCategory::try_from(filter.as_ref()) {
+                Ok(category) => Ok(LintFilterKind::Category(category)),
+                Err(()) => Ok(LintFilterKind::Generic(filter)),
+            }
+        }
+    }
+}
+
+impl TryFrom<String> for LintFilterKind {
+    type Error = InvalidFilterKind;
+
+    #[inline]
+    fn try_from(filter: String) -> Result<Self, Self::Error> {
+        Self::parse(Cow::Owned(filter))
+    }
+}
+
+impl TryFrom<&'static str> for LintFilterKind {
+    type Error = InvalidFilterKind;
+
+    #[inline]
+    fn try_from(filter: &'static str) -> Result<Self, Self::Error> {
+        Self::parse(Cow::Borrowed(filter))
+    }
+}
+
+impl TryFrom<Cow<'static, str>> for LintFilterKind {
+    type Error = InvalidFilterKind;
+
+    #[inline]
+    fn try_from(filter: Cow<'static, str>) -> Result<Self, Self::Error> {
+        Self::parse(filter)
+    }
+}
+
+impl From<RuleCategory> for LintFilterKind {
+    #[inline]
+    fn from(category: RuleCategory) -> Self {
+        LintFilterKind::Category(category)
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum InvalidFilterKind {
```

Comment:
We can use `thiserror`.

Reply:
Both `miette` and `thiserror` were removed from the linter's dependencies in favor of `oxc_diagnostics`. This crate does not re-export anything from `thiserror`. Are you OK with me exporting `thiserror`'s macros from this crate?

Reply:
Oh ... I removed `thiserror` everywhere.